force-app/main/default/classes/RD2_OpportunityEvaluationService.cls

```diff
@@ -351,6 +366,8 @@ public inherited sharing class RD2_OpportunityEvaluationService {
         List<npe03__Recurring_Donation__c> closeOppRds = new List<npe03__Recurring_Donation__c>();
         List<RD2_RecurringDonation> rdRecords = new List<RD2_RecurringDonation>();
 
+        setOldRdListIfNull();
```

Comment:
This turned out to be the fix when using the Pause modal window, since the actual update happens in `handleOutdatedRecurringDonations` further in the code (rather than in a trigger), I needed a way to compare the changed RDs to their original form.